src/main/java/io/wisetime/connector/jira/database/JiraDb.java

```diff
@@ -35,32 +55,126 @@ public void asTransaction(final Runnable runnable) {
+    );
+
+    requiredTablesAndColumnsMap.put(
+        "propertystring",
+        ImmutableSet.of("id", "propertyvalue")
+    );
+
+
+    Map<String, List<String>> actualTablesAndColumnsMap = query.databaseInspection()
+        .selectFromMetaData(meta -> meta.getColumns(null, null, null, null))
+        .listResult(rs -> ImmutablePair.of(rs.getString("TABLE_NAME"), rs.getString("COLUMN_NAME")))
+        .stream()
+        .filter(pair -> requiredTablesAndColumnsMap.containsKey(pair.getKey().toLowerCase()))
+        .collect(groupingBy(ImmutablePair::getKey, mapping(ImmutablePair::getValue, toList())));
+
+    return requiredTablesAndColumnsMap.entrySet().stream()
+        // Values from MetaDataResultSet are in uppercase
+        .allMatch(entry -> actualTablesAndColumnsMap.containsKey(entry.getKey().toUpperCase()) &&
+                actualTablesAndColumnsMap.get(entry.getKey().toUpperCase()).containsAll(
+                    entry.getValue().stream().map(String::toUpperCase).collect(Collectors.toSet()))
+        );
+  }
+
+  public boolean hasConfiguredTimeZone() {
+    try {
+      getJiraDefaultTimeZone();
+      // If above query did not fail, it means we can connect to Jira DB and we can insert worklog with correct timezone
+      return true;
+    } catch (Exception ex) {
+      return false;
+    }
   }
 
   public Optional<Issue> findIssueByTagName(final String tagName) {
-    // TODO
-    return Optional.empty();
+    Optional<Pair<String, Integer>> projectIssuePair = getJiraProjectIssuePair(tagName);
+
+    if (!projectIssuePair.isPresent()) {
+      return Optional.empty();
+    }
+
+    return query.select("SELECT jiraissue.id, project.pkey, jiraissue.issuenum, jiraissue.summary, jiraissue.timespent "
+            + "FROM project INNER JOIN jiraissue ON project.id = jiraissue.project "
+            + "WHERE project.pkey = ? AND jiraissue.issuenum = ?")
+        .params(
+            projectIssuePair.get().getLeft(),
+            projectIssuePair.get().getRight()
+        )
+        .firstResult(this::buildIssueFromResultSet);
   }
 
   public List<Issue> findIssuesOrderedById(final long startIdExclusive, final int maxResults) {
-    // TODO
-    return ImmutableList.of();
+    return query.select("SELECT jiraissue.id, project.pkey, jiraissue.issuenum, jiraissue.summary, jiraissue.timespent "
+        + "FROM project INNER JOIN jiraissue ON project.id = jiraissue.project "
+        + "WHERE jiraissue.id > ? ORDER BY ID ASC LIMIT ?;")
+        .params(
+            startIdExclusive,
+            maxResults
+        )
+        .listResult(this::buildIssueFromResultSet
+        );
   }
 
   public Optional<String> findUsername(final String email) {
-    // TODO
-    return Optional.empty();
+    return query.select("SELECT user_name FROM cwd_user WHERE lower_email_address = :email")
+        .namedParam("email", email.toLowerCase())
+        .firstResult(resultSet -> resultSet.getString(1));
```

Comment:
FluentJdbc has shortcut for this usecase: 
```suggestion
        .firstResult(Mappers.singleString());
```
Note: new import required:
```java
import org.codejargon.fluentjdbc.api.mapper.Mappers;
```